### PR TITLE
Fix indentation for extensions dashboards

### PR DESCRIPTION
**Pull request by MartinWeindel**
**How to categorize this PR?**

/area control-plane
/area monitoring
/kind bug

**What this PR does / why we need it**:
All extensions dashboards are not visible anymore in Grafana for both operators and users.
The bug is a wrong indent in the template.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
For an old shoot, you can verify in the control plane namespace that all extensions dashboards are missing since some time:

```bash
$ k get cm | grep dashboards
grafana-operators-dashboards                     30     88d
grafana-operators-dashboards-46adb4b0            24     15d
grafana-users-dashboards                         16     88d
grafana-users-dashboards-574117d5                13     15d
```

**Release note**:

```bugfix operator
Fix indentation for extensions Grafana dashboards
```

### CODE REVIEW

File: charts/seed-monitoring/charts/grafana/templates/_helpers.tpl [hunk 1]
```diff
@@ -82,7 +82,7 @@ grafana-{{ .Values.role }}-datasources-{{ include "grafana.datasources.data" . |
 {{ end }}
 {{ end }}
 {{- if .Values.extensions.dashboards }}
-{{- toString .Values.extensions.dashboards | indent 2 }}
+{{- toString .Values.extensions.dashboards }}
```
Comment (timebertt) [inline charts/seed-monitoring/charts/grafana/templates/_helpers.tpl:85]: I'm not sure, if I understand how a changed indentation fixes "missing" values.
Can you elaborate on what the bug exactly was?

Comment (MartinWeindel) [reply charts/seed-monitoring/charts/grafana/templates/_helpers.tpl:85]: The data items for the extensions dashboards have been added to the yaml with indentation of 4 spaces.

For example:
```
apiVersion: v1
data:
  apiserver-admission-details.json: |-
…
  vpn-dashboard.json: |-
    {
…
    }
 
    calico-felix-dashboard.json: |-
      {
…
      }
```

Here `apiserver-admission-details.json` and `vpn-dashboard.json` are predefined dashboards and correctly indented. But the extension dashboard `calico-felix-dashboard.json` is indented with 4 spaces.
The predefined dashboards are added with two parts. The `{{ base $name }}` is added without indent and only the value part is indented. The error is that the extension dashboard is added as one string consisting of both key and value and therefore needs no indentation.

**Review by ialidzhikov — APPROVED** [review]
Thank you for this PR!
Some time ago I also noticed that for example calico dashboards were no longer present in the operators grafana. Back then I noted down this to follow-up but I was not able to get into it.
I now verified locally that with this PR the calico dashboards are available in the operators grafana.

/lgtm

File: charts/seed-monitoring/charts/grafana/templates/_helpers.tpl (hunk 1 above)
Comment (timebertt) [reply charts/seed-monitoring/charts/grafana/templates/_helpers.tpl:85]: Ahh, thanks for explaining. That makes sense :)

**Review by timebertt — APPROVED** [review]
/lgtm
/milestone v1.36

**Comment by timebertt** [comment]
If I see it correctly, this is broken since https://github.com/gardener/gardener/pull/4528
Hence, we should cherry-pick this PR to the last 3 releases.

**Pull request merged**